File: Backend.js
const express = require('express');
const app = express();

// Middleware pour parser les données JSON du corps de la requête
app.use(express.json());

// Route pour recevoir les données du script Python
app.post('/api/receive_data', (req, res) => {
    const { username, followers_count } = req.body;

    if (!username || !followers_count) {
        return res.status(400).json({ message: "Le nom d'utilisateur et le nombre d'abonnés sont requis." });
    }

    // Affiche les données reçues dans la console
    console.log(`Données reçues: Nom d'utilisateur - ${username}, Nombre d'abonnés - ${followers_count}`);

    // Répond au client que les données ont été reçues avec succès
    res.status(200).json({ message: "Données reçues avec succès." });
});

// Démarre le serveur sur le port 5000
const PORT = 5000;
app.listen(PORT, () => {
    console.log(`Le serveur écoute sur le port ${PORT}`);
});

File: Backend.py
from flask import Flask, request, jsonify

app = Flask(__name__)

@app.route('/api/receive_data', methods=['POST'])
def receive_data():
    # Récupère les données JSON envoyées
    data = request.get_json()

    if not data:
        return jsonify({"error": "No data provided"}), 400

    username = data.get("username")
    followers_count = data.get("followers_count")

    if username is None or followers_count is None:
        return jsonify({"error": "Missing username or followers_count"}), 400

    # Traitement des données (ici, on les affiche simplement)
    print(f"Received data - Username: {username}, Followers Count: {followers_count}")

    # Réponse au client
    return jsonify({"status": "success", "message": "Data received successfully"}), 200

if __name__ == '__main__':
    app.run(host='0.0.0.0', port=5000)
